app/routes.py

```diff
@@ -1,97 +1,232 @@
+def add_harvest_error():
+    if request.is_json:
+        error = db.add_harvest_error(request.json)
+        if error:
+            return jsonify({"message": f"Added new harvest error with ID: {error.id}"})
+        else:
+            return jsonify({"error": "Failed to add harvest error."}), 400
+    else:
+        return jsonify({"Please provide harvest error with json format."})
+
+@mod.route("/harvest_error/", methods=["GET"])    
+@mod.route("/harvest_error/<error_id>", methods=["GET"])
+def get_harvest_error(error_id=None):
     try:
-        result = db.get_harvest_job(job_id)
-        return result
+        if error_id:
+            error = db.get_harvest_error(error_id)
+            return jsonify(error) if error else ("Not Found", 404)
+
+        job_id = request.args.get("harvest_job_id")
+        if job_id:
+            error = db.get_harvest_error_by_job(job_id)
+            if not error:
+                return "No harvest errors found for this harvest job", 404
+        else:
+            # for test, will remove later
+            error = db.get_all_harvest_errors()
+    
+        return jsonify(error)
     except Exception:
-        return "provide job_id"
+        return "Please provide correct error_id or harvest_job_id"
 
-@mod.route('/harvest_error/<error_id>', methods=['GET'])
-def get_harvest_error(error_id):
+@mod.route("/harvest_record/add", methods=["POST", "GET"])
+def add_harvest_record():
+    if request.is_json:
+        record = db.add_harvest_record(request.json)
+        if record:
+            return jsonify({"message": f"Added new record with ID: {record.id}"})
+        else:
+            return jsonify({"error": "Failed to add harvest record."}), 400
+    else:
+        return jsonify({"Please provide harvest record with json format."})
+
+@mod.route("/harvest_record/", methods=["GET"])    
+@mod.route("/harvest_record/<record_id>", methods=["GET"])
+def get_harvest_record(record_id=None):
     try:
-        result = db.get_harvest_error(error_id)
-        return result
+        if record_id:
+            record = db.get_harvest_record(record_id)
+            return jsonify(record) if record else ("Not Found", 404)
+
+        job_id = request.args.get("harvest_job_id")
+        source_id = request.args.get("harvest_source_id")
+        if job_id:
+            record = db.get_harvest_record_by_job(job_id)
+            if not record:
+                return "No harvest records found for this harvest job", 404
+        elif source_id:
+            record = db.get_harvest_record_by_source(source_id)
+            if not record:
+                return "No harvest records found for this harvest source", 404
+        else:
+            # for test, will remove later
+            record = db.get_all_harvest_records()
+    
+        return jsonify(record)
     except Exception:
-        return "provide error_id"
+        return "Please provide correct record_id or harvest_job_id"
```

Comment:
remove `harvest_job_id`

Reply:
since we may get job either with record_id or job_id, the message is for both possible situations.